From 2b0c3626b1ba23f6a2b435cab218c708a647c777 Mon Sep 17 00:00:00 2001
From: frq09432 <nicolas.louboutin@st.com>
Date: Fri, 11 Oct 2019 13:49:59 +0200
Subject: [PATCH 36/38] binder failed_at available only in case of failure

---
 drivers/android/binder.c | 17 ++++++++++++-----
 1 file changed, 12 insertions(+), 5 deletions(-)

diff --git a/drivers/android/binder.c b/drivers/android/binder.c
index 71e777d..7717b79 100644
--- a/drivers/android/binder.c
+++ b/drivers/android/binder.c
@@ -2422,11 +2422,18 @@ static void binder_transaction_buffer_release(struct binder_proc *proc,
 	int debug_id = buffer->debug_id;
 	binder_size_t off_start_offset, buffer_offset, off_end_offset;
 
-	binder_debug(BINDER_DEBUG_TRANSACTION,
-		     "%d buffer release %d, size %zd-%zd, failed at %llx\n",
-		     proc->pid, buffer->debug_id,
-		     buffer->data_size, buffer->offsets_size,
-		     (unsigned long long)failed_at);
+	if (is_failure) {
+		binder_debug(BINDER_DEBUG_TRANSACTION,
+				"%d buffer release %d, size %zd-%zd, failed at %llx\n",
+				proc->pid, buffer->debug_id,
+				buffer->data_size, buffer->offsets_size,
+				(unsigned long long)failed_at);
+	} else {
+		binder_debug(BINDER_DEBUG_TRANSACTION,
+				"%d buffer release %d, size %zd-%zd\n",
+				proc->pid, buffer->debug_id,
+				buffer->data_size, buffer->offsets_size);
+	}
 
 	if (buffer->target_node)
 		binder_dec_node(buffer->target_node, 1, 0);
-- 
2.7.4

