From 66d862823c53be6e43dc117631cbc1320e3d8b04 Mon Sep 17 00:00:00 2001
From: Yves COPPEAUX <yves.coppeaux@st.com>
Date: Wed, 11 Oct 2023 10:52:16 +0200
Subject: [PATCH 02/23] ANDROID: binder: failed_at available only in case of
 failure

---
 drivers/android/binder.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/android/binder.c b/drivers/android/binder.c
index 4ea9a4dae15f5..c8aa89d4c5a85 100644
--- a/drivers/android/binder.c
+++ b/drivers/android/binder.c
@@ -2140,11 +2140,18 @@ static void binder_transaction_buffer_release(struct binder_proc *proc,
 	int debug_id = buffer->debug_id;
 	binder_size_t off_start_offset, buffer_offset;
 
+	if (is_failure) {
 	binder_debug(BINDER_DEBUG_TRANSACTION,
 		     "%d buffer release %d, size %zd-%zd, failed at %llx\n",
 		     proc->pid, buffer->debug_id,
 		     buffer->data_size, buffer->offsets_size,
 		     (unsigned long long)off_end_offset);
+	} else {
+		binder_debug(BINDER_DEBUG_TRANSACTION,
+				"%d buffer release %d, size %zd-%zd\n",
+				proc->pid, buffer->debug_id,
+				buffer->data_size, buffer->offsets_size);
+	}
 
 	if (buffer->target_node)
 		binder_dec_node(buffer->target_node, 1, 0);
-- 
2.34.1

